docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
+[**`Exceptions`**](#exceptions-classes) represents a collection of exceptions used by multiple components above.
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+<img src="images/ArchitectureSequenceDiagram.png" alt=""/>
+
+The sections below give more details of each component.
+
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
+
+### Parser component
+
+**API** : [`Parser.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/parser/Parser.java)
+
+![Structure of the Parser Component](images/ParserClassDiagram.png)
+
+How the parsing works:
+1. When user inputs, the input is passed to the `Parser`.
+2. `Parser` first extracts the command.
+3. Using the extracted command, `Parser` will perform different validation checks on the arguments supplied in the
+input.
+4. Arguments (mainly the `pid`, `name`, `quantity`, `price`, `description`, `amount` fields) are validated. 
+Exceptions are thrown when the fields do not pass their respective type checks.
+5. Once validation passes, `Parser` uses the validated arguments to creates an instance of that particular command. 
+For example, a `delete` command will cause `Parser` to create a new instance of `DeleteCommand(pid)`.
+6. The created command object is returned back to `main` function for further processing.
+
+### Commands component
+
+**API** : [`Command.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/Command.java)
+
+
+![Structure of the Commands Component](images/CommandsClassDiagram.png)
+
+### Data component
+
+**API** : [`Data`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/data)
+
+The following is a class diagram of the `data` component.
+![Structure of the Data Component](images/DataClassDiagram.png)
+
+
+### Storage component
+
+**API** : [`Storage.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/storage/Storage.java)
+
+![Structure of the Storage Component](images/StorageClassDiagram.png)
+
+The `Storage` component,
+* can save product list data in CSV format, and load them back into corresponding Products.
+* depends on the `StockPal` component (because the `Storage` component's job is to save/load objects that belong to `StockPal`)
+* consists of the classes `Storage`, `CsvWriter` and `CsvReader`.
+  * `Storage` defines methods that loads and saves data.
+  * `CsvWriter` is responsible for handling the writing of data to the CSV data file.
+  * `CsvReader` is responsible for handling the reading of data from the CSV data file.
+
+### Common classes
```

Comment:
The explanation of common classes is too brief, can give more detailed explanation them and provide some examples